src/collection.ts

```diff
@@ -693,8 +701,13 @@ export class Collection<TSchema extends Document = Document> {
    *
    * @param options - Optional settings for the command
    */
-  async indexInformation(options?: IndexInformationOptions): Promise<Document> {
-    return await this.indexes({ ...options, full: options?.full ?? false });
+  async indexInformation<TFull extends boolean = false>(
```

Comment:
It's preferrable to avoid computed types as the return types from our APIs, since they don't provided users with information about the return type of a function until a user 1) reads the source of the computed type or 2) writes code using the API and checks the computed return type.  Could we accomplish the same here using overloads and removing all `TFull` generics?  Something like:

```typescript
  indexInformation(
    options: IndexInformationOptions & { full: true }
  ): Promise<IndexDescriptionInfo[]>;
  indexInformation(
    options: IndexInformationOptions & { full: false }
  ): Promise<IndexDescriptionCompact>;
  indexInformation(
    options: IndexInformationOptions & { full: boolean }
  ): Promise<IndexDescriptionCompact | IndexDescriptionInfo[]>;
```

This has the additional benefits of 

- better documentation, because our tooling will generate a doc entry for each overload
- clearly defined overloads with existing IDE tooling
- each overload can be documented originally (if desired).

for example - here's what VSCode could hypothetically display if we used overloads:

<img width="793" alt="Screenshot 2024-04-09 at 12 23 01 PM" src="https://github.com/mongodb/node-mongodb-native/assets/23407842/ac95d9f5-07f1-4a29-984a-786eeb2bd2b6">
<img width="633" alt="Screenshot 2024-04-09 at 12 22 55 PM" src="https://github.com/mongodb/node-mongodb-native/assets/23407842/0e6c8042-c045-4f5f-8247-7440d3af9d46">